public/scripts/stackedbar-util.js

```diff
@@ -1,15 +1,83 @@
 ////////////////////////////// For running time stacked bar ///////////////////
-var StackedBarUtil = (function($) {
-  var sbutil = {};
+var StackedBarUtil = (function($, StateUtil) {
```

Comment:
I'm just going to leave as-is for now.  Nothing is really coming to mind, other than factoring out the shared left/right toggle code as a future task.